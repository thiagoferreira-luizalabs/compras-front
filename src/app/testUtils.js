import React from "react"
import { configure as enzymeConfigure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from "sinon";
import { MemoryRouter } from "react-router-dom";

enzymeConfigure({ adapter: new Adapter() });

const asyncAssert = (callback, done) => {
  setTimeout(() => {
    try {
      callback()
      done()
    } catch (e) {
      done.fail(e)
    }
  }, 5)
}

const createFileChangeEvent = (file) => (
  {
    target: {
      files: [file]
    }
  }
)

const MemoryRouterHOC = (children) => (<MemoryRouter>{children}</MemoryRouter>)

const toJsonSnapshot = (wrapper) => toJson(wrapper, { mode: "deep", noKey: true })

export {
  shallow, mount, sinon, MemoryRouter, toJsonSnapshot, asyncAssert, createFileChangeEvent, MemoryRouterHOC
}
