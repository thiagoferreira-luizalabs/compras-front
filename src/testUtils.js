import { configure as enzymeConfigure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from "sinon";
// import { MemoryRouter } from "react-router-dom";

enzymeConfigure({ adapter: new Adapter() });

const asyncAssert = (callback, done) => {
  setTimeout(() => {
    try {
      callback()
      done()
    } catch (e) {
      done.fail(e)
    }
  }, 5)
}

const createFileChangeEvent = (file) => (
  {
    target: {
      files: [file]
    }
  }
)

export {
  shallow, mount, sinon,  /*MemoryRouter,*/ toJson, asyncAssert, createFileChangeEvent
}
