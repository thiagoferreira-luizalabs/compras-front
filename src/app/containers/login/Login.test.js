import React from 'react';
import Login from "./Login"
import { mount, toJsonSnapshot } from "../../testUtils"

it("Renders without crashing", () => {
  let wrapper = mount(<Login />)
  expect(toJsonSnapshot(wrapper)).toMatchSnapshot()
})