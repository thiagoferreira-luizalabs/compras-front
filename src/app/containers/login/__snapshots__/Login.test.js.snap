// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
<div
  className="login"
>
  <div
    className="box"
  >
    <div
      className="ui grid"
    >
      <div
        className="computer only six wide column brand"
      >
        <img
          className="ui image logo"
          src="magazineluiza.png"
        />
        <span>
          Acesse sistemas, e administrativos tudo em um s칩 lugar.
        </span>
        <img
          className="ui image lu"
          src="lu.png"
        />
      </div>
      <div
        className="ten wide computer sixteen wide mobile column form"
      >
        <form
          className="ui form"
          onSubmit={[Function]}
        >
          <h1>
            Acesso ao Sistema
          </h1>
          <div
            className="field"
          >
            <label>
              Usu치rio
            </label>
            <input
              placeholder="Digite seu usu치rio"
            />
            <span>
              Usu치rio do magazineluiza
            </span>
          </div>
          <div
            className="field"
          >
            <label>
              Senha
            </label>
            <input
              placeholder="Digite sua senha"
            />
            <span>
              Senha do portal
            </span>
          </div>
          <button
            className="ui green fluid button"
            onClick={[Function]}
            role="button"
            type="submit"
          >
            Acessar
          </button>
          <span
            className="forgot-password"
          >
            Esqueceu sua senha? 
            <a
              href="/"
            >
              Clique aqui
            </a>
          </span>
        </form>
      </div>
    </div>
  </div>
</div>
`;
